### ISSUE TITLE
Can not get target.page() for webviews inside Electron browser

### ISSUE THREAD

**Issue by JustMyst** [puppeteer/puppeteer#6473]
I have an Electron application, that uses two webviews inside main page, and I'm trying to test it with external tool, that connects to Electron via Websocket. There is no problem with connection (I am able to connect to electron browser), but I'm stuck in getting pages of MainPage and Webviews in same time.

### Steps to reproduce

**Environment:**

* Puppeteer version: 5.2.1
* Platform / OS version: Windows 10
* Node.js version: 12.16.3
* Electron version: 9.3.1

**What steps will reproduce the problem?**

1. Create page with two webviews.
```
        <div class="webview-container">
            <webview plugins webpreferences="nativeWindowOpen=true" allowpopups>
            </webview>
            <webview plugins webpreferences="nativeWindowOpen=true" allowpopups>
            </webview>
        </div>
```
Webview source is written in script loaded at the end of file
```
this._webviews = <HTMLElement[]>Array.from(this._document.getElementsByTagName("webview"));

    this._webviews.map((e, i) => {
      e.setAttribute("src", path[i]);
      e.id = name[i];
    });
``` 
2. Include it in Electron App and append 
```
    app.commandLine.appendSwitch(
        'remote-debugging-port',
        '8313'
    );
```
Or run it in Chrome with remote debugging port opened.
3. Connect with Electron or chrome with another node app
```
        const response = await fetch(`http://localhost:8313/json/version`);
        const debugEndpoints = await response.json();

        const webSocketDebuggerUrl = debugEndpoints.webSocketDebuggerUrl;

        const browser = await puppeteer.connect({
            browserWSEndpoint: webSocketDebuggerUrl,
            defaultViewport: {
                hasTouch: true,
                width: 1920,
                height: 1080
            }
        });
``` 

**What is the expected result?**
At this moment I thought I could get all `page`s  by using `browser.pages()`, but it does return only MainPage - No webviews.

Other way - to use `browser.targets()` and get `target.page()` for each target to get proper page.

**What happens instead?**

As said - `browser.pages()` returns no Webview.

`browser.target()` returns 4 targets - Browser, Page, and 2 Webviews (Correctly!) **but**:
* using `MainPageTarget.page()` returns MainPage ( OK ) and reloads whole page with webviews, therefore `webviewTargets` lose connection and can no longer get `webviewTarget.page()` (Throws error, that target is already destroyed)
* using `webviewTarget.page()` and then `MainPageTarget.page()` throws error, because again MainPage is reloaded and kills webviews.
* waiting for webviewTarget with `browser.waitForTarget()` does not help either. After waiting a while webview targets seem to disapear
```
        const targets = await browser.targets(); // returns 4 targets
        const mainPage = await targets.find(e => e.type() === "page").page(); // get MainPage from its target

        const frames1 = mainPage.frames(); // returns 3 frames (page and 2 webviews with "about:blank" in url)

        const webviewTarget = await browser.waitForTarget(e => <any>e.type() === "webview"); // waits for next webview target. This one has different targetId than those in first line

        const frames2 = mainPage.frames(); // returns only one frame leading to page. No webview frames.
```
* using `MainPage.frames()` returns 3 frames - Page and Webviews (OK) but those webviews point to "about:blank" - do not have any name, nor URL.

**My questions here**:
1. Does loading `target.page()` when connected to external chromium **must** reload whole page or is this configurable? I believe this causes `WebviewTargets` to crush while getting its page.
2. Is there any proper order in which I should get Pages/frames when using page with webviews in it?
3. Is this how it should work, or is it a bug with webviews?

**Comment by sirdir** [comment]
@JustMyst did you find the solution? looks like I have a similar problem.

**Comment by sirdir** [comment]
It not working with puppeteer-core. 

But! I just tried with Selenium Chrome WebDriver to connect to the same application. And if you specify in chromeoptions windowTypes as webview, it starts to discover it as a separate page. 

After that I used puppeteer to communicate with app via devtools protocol but [Target.getTargets](https://chromedevtools.github.io/devtools-protocol/tot/Target/#method-getTargets) didn't return webview.
